qemu/tests/virtio_console.py

```diff
@@ -1792,19 +1765,31 @@ def test_unplugged_restart():
                       method=params.get('virtio_console_method', 'shell'),
                       timeout=720)
         except Exception, details:
-            raise error.TestFail("Fail to reboot VM:\n%s" % details)
+            test.fail("Fail to reboot VM:\n%s" % details)
 
         # TODO: Hotplug ports and verify that they are usable
         # VM is missing ports, which are in params.
         vm.destroy(gracefully=True)
 
-    @error.context_aware
+    @error_context.context_aware
     def test_failed_boot():
         """
         Start VM and check if it failed with the right error message.
         :param cfg: virtio_console_params - Expected error message.
         """
-        exp_error_message = params['virtio_console_params']
+        exp_error_message = (params['virtio_console_params'] % 31)
```

Comment:
I think it may be better put these numbers（32，31）（512，511） into cfg and give a meaningful name. 
What about put this line into the else clause under line 1791? are they always minus one? 
And I noticed doc string of this function is outdated, can you please also update it please?

Reply:
Thanks for advice. I separated these nums to valid set and invalid set. (512, 32) are the minimum invalid boundary cases for `max_ports`, so they always be the valid `max_ports` + 1. 
I updated my code again and could you help review it the third time? Thanks!

RHEV7:
smp_2.2048m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.myprovider.virtio_console.spread_linear.incorrect_options.boot_too_much_ports: PASS (6.47 s)
RHEL7:
smp_2.2048m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.myprovider.virtio_console.spread_linear.incorrect_options.boot_too_much_ports: PASS (6.16 s)